toolkit/tools/pkg/imagecustomizerlib/cosicommon.go

```diff
@@ -303,3 +306,38 @@ func getArchitectureForCosi() string {
 	}
 	return runtime.GOARCH
 }
+
+func getAllPackagesFromChroot(imageConnection *ImageConnection) ([]OsPackage, error) {
+	var out string
+	err := imageConnection.Chroot().UnsafeRun(func() error {
+		var err error
+		out, _, err = shell.Execute(
+			"rpm", "-qa", "--queryformat", "%{NAME} %{VERSION} %{RELEASE} %{ARCH}\n",
+		)
+		if err != nil {
+			return fmt.Errorf("failed to list installed RPMs:\n%w", err)
+		}
+		return nil
```

Comment:
You already have a `fmt.Errorf` outside the `UnsafeRun` call. So, you don't need one inside the `UnsafeRun` call.